app/services/hmrc_interface_result_service.rb

```diff
@@ -0,0 +1,32 @@
+class HmrcInterfaceResultService
+  attr_reader :submission
+
+  def self.call(*args)
+    new(*args).call
+  end
+
+  def initialize(submission_id)
+    @submission = Submission.find(submission_id)
+  end
+
+  def call
+    submission.update!(status: "completing")
+
+    response = HmrcInterface::Request::Result.call(client, submission.hmrc_interface_id)
+
+    submission.update!(hmrc_interface_result: response, status: response[:status])
+
+    # "processing" and "created" statuses require a retry
+    unless %w[completed failed].include?(response[:status])
+      raise HmrcInterface::TryAgain,
```

Comment:
I wasn't sure about raising an error here, as to some extent this could be expected behaviour i.e. we know the result will not be available immediately. I know that we have used this to trigger a retry. Do we use this pattern elsewhere?

Reply:
The sentry errors only catches when retries have been exhausted. Ideally a response should never be exhausted and it would be good to see how many we get to start with. If it gets very noisy in production we can exclude I'd say